enhancements/testing/openshift-tests-extension.md

```diff
@@ -0,0 +1,668 @@
+tests the binary offers can be discovered and executed. This interface includes several command line
+verbs the binary must implement, arguments those verbs must support, and output formats the binary
+must adhere to.
+
+Running an extension binary will output the following help text for the initial version of the interface. 
+```
+info      - Output test contribution extension version and metadata.
+list      - Output tests supported by this extension.
+run-test  - Run one or more tests and output results.
+config    - Component configuration management.
+```
+
+##### Info - Extension Metadata
+The extension interface will evolve over time. For this reason, the `info` verb
+will output information about which version of the interface the binary supports. The origin
+framework will support some level of skew in the interface versions and invoke the binaries
+with verbs & arguments consistent with the interface version they support.
+
+Info also provides information to origin about how to construct new or contribute new
+test suites.
+
+Annotated example `info` output is provided below.
+```python
+{
+    # The extension interface version the binary implements.
+    # The details of the interface and version will usually be
+    # unnecessary for a contributor to understand as they will
+    # vendor in the majority of the implementation from the
+    # "Test Extension Support Module" explained later.
+    "apiVersion": "1.0",
+
+    "source": {
+        # The commit from which this binary was compiled.
+        "commit": "87fdbaa",
+        # The git repository (if known) that this
+        # extension was built from.
+        "source_url": "github.com/openshift/kubernetes"
+    },
+
+    # A single extension can carry tests for multiple
+    # components. However, each invocation of the binary
+    # can only represent and act on information relative
+    # to a single component.
+    # If a binary carries tests for multiple components,
+    # it must register itself multiple times, with
+    # different pre-verb arguments which openshift-tests
+    # will provide verbatim with every invocation; e.g.
+    # if the binary registers with "--component hyperkube",
+    # verb invocations will be as follows:
+    #    extension-binary --component hyperkube info
+    #    extension-binary --component hyperkube list ...
+    "component": {
+        "product": "openshift",
+        "type": "payload",
+        "name": "hyperkube"
+    },
+
+    "configurations": [
+        {
+            "profile": "common-mode-1",
+            "description": "Enables a common, non-default mode of operation where ....",
+            "application": {
+                # If applying the configuration implies a disruption, inform 
+                # openshift-tests, so that it can be accounted for in overall
+                # disruption reporting.
+                "disruption": "1m",

```

Comment:
This makes me a little nervous if it's happening in other repos and we didn't have good visibility into people abusing this because a problem popped up. Probably not a core concern for this enhancement though.